packages/dataviews/src/add-filter.js

```diff
@@ -145,105 +155,73 @@ export default function AddFilter( { filters, view, onChangeView } ) {
-												role="menuitemradio"
-												aria-checked={
-													activeOperator ===
-													OPERATOR_IN
-												}
-												prefix={
-													activeOperator ===
-														OPERATOR_IN && (
-														<Icon icon={ check } />
-													)
-												}
-												onSelect={ ( event ) => {
-													event.preventDefault();
-													onChangeView( {
-														...view,
-														page: 1,
-														filters: [
-															...otherFilters,
-															{
-																field: filter.field,
-																operator:
-																	OPERATOR_IN,
-																value: filterInView?.value,
-															},
-														],
-													} );
-												} }
-											>
-												{ __( 'Is' ) }
-											</DropdownMenuItem>
-											<DropdownMenuItem
-												key="not-in-filter"
-												role="menuitemradio"
-												aria-checked={
-													activeOperator ===
-													OPERATOR_NOT_IN
-												}
-												prefix={
-													activeOperator ===
-														OPERATOR_NOT_IN && (
-														<Icon icon={ check } />
-													)
-												}
-												onSelect={ ( event ) => {
-													event.preventDefault();
-													onChangeView( {
-														...view,
-														page: 1,
-														filters: [
-															...otherFilters,
-															{
-																field: filter.field,
-																operator:
-																	OPERATOR_NOT_IN,
-																value: filterInView?.value,
-															},
-														],
-													} );
-												} }
-											>
-												{ __( 'Is not' ) }
-											</DropdownMenuItem>
-										</DropdownSubMenu>
+											{ Object.entries( OPERATORS ).map(
+												( [
+													operator,
+													{ label, key },
+												] ) => (
+													<DropdownMenuRadioItem
+														key={ key }
+														name={ `add-filter-${ filter.name }-conditions` }
+														value={ operator }
+														checked={
+															activeOperator ===
+															operator
+														}
+														onChange={ () => {
+															onChangeView( {
+																...view,
+																page: 1,
+																filters: [
+																	...otherFilters,
+																	{
+																		field: filter.field,
+																		operator,
+																		value: filterInView?.value,
+																	},
+																],
+															} );
+														} }
+													>
+														<DropdownMenuItemLabel>
+															{ label }
+														</DropdownMenuItemLabel>
+													</DropdownMenuRadioItem>
+												)
+											) }
+										</DropdownMenu>
 									) }
 									<DropdownMenuItem
 										key={ 'reset-filter-' + filter.name }
 										disabled={ ! activeElement }
-										onSelect={ ( event ) => {
+										onClick={ ( event ) => {
 											event.preventDefault();
```

Comment:
I think we can remove this now. It was an issue with the old component.

Reply:
Removing this line would cause the dropdown to close when clicking this menu item — would that be ok?

Reply:
We want to keep the "AddFilter" dropdown open https://github.com/WordPress/gutenberg/pull/56983#issuecomment-1852338071

Reply:
In general, I think this PR should be as close to a refactor as possible: no change in behavior, just change the underlying implementation.

Reply:
Maybe use the `hideOnClick` prop then?

Reply:
Thank you for explaining the desidered behavior — `hideOnClick` should be the correct way to implement it, as Nik suggests. I'll go ahead and make the changes

Reply:
https://github.com/WordPress/gutenberg/pull/55625/commits/0520eb246d1586c4a205efcdda21926c0177ad3c

Reply:
I've prepared https://github.com/WordPress/gutenberg/pull/57418 to close the actions menu upon selecting the layout. At some point in this PR this behavior was removed.